pkg/util/installinfo/install_info.go

```diff
@@ -135,3 +147,97 @@ func logVersionHistoryToFile(versionHistoryFilePath, installInfoFilePath, agentV
 		return
 	}
 }
+
+// WriteInstallInfo write install info and signature files
+func WriteInstallInfo(installerVersion, installType string) error {
+	// avoid rewriting the files if they already exist
+	if _, err := os.Stat(installInfoFile); err == nil {
+		log.Info("Install info file already exists, skipping")
+		return nil
+	}
+	tool, version := getToolVersion()
+	if err := writeInstallInfo(tool, version, installerVersion); err != nil {
+		return fmt.Errorf("failed to write install info file: %v", err)
+	}
+	if err := writeInstallSignature(installType); err != nil {
+		return fmt.Errorf("failed to write install signature file: %v", err)
+	}
+	return nil
+}
+
+// RmInstallInfo removes the install info and signature files
+func RmInstallInfo() {
+	if err := os.Remove(installInfoFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install info file: %s", err)
+	}
+	if err := os.Remove(installSigFile); err != nil && !os.IsNotExist(err) {
+		log.Warnf("Failed to remove install signature file: %s", err)
+	}
+}
+
+func getToolVersion() (string, string) {
+	tool := "unknown"
+	version := "unknown"
+	if _, err := exec.LookPath("dpkg-query"); err == nil {
+		tool = "dpkg"
+		toolVersion, err := getDpkgVersion()
+		if err == nil {
+			version = toolVersion
+		}
+	}
+	if _, err := exec.LookPath("rpm"); err == nil {
+		tool = "rpm"
+		toolVersion, err := getRPMVersion()
+		if err == nil {
+			version = fmt.Sprintf("rpm-%s", toolVersion)
+		}
+	}
+	return tool, version
+}
+
+func getRPMVersion() (string, error) {
+	output, err := exec.Command("rpm", "-q", "-f", "/bin/rpm", "--queryformat", "%%{VERSION}").Output()
```

Comment:
NB: there's no e2e test yet checking the RPM file, only deb
It will be tested soon once we add support for RPM for the updater (next 2 weeks)